addons/html_builder/static/tests/background_shape_color.test.js

```diff
@@ -0,0 +1,267 @@
+import { expect, test, describe } from "@odoo/hoot";
+import { animationFrame, queryOne } from "@odoo/hoot-dom";
+import { contains } from "@web/../tests/web_test_helpers";
+
+describe.current.tags("desktop");
+
+function getShapeTestCSS() {
+    return `
+        /* CSS Variables for theme colors */
+        :root {
+            --o-color-1: #3aadaa;
+            --o-color-2: #7c6576;
+            --o-color-3: #dd7777;
+            --o-color-4: #383e45;
+            --o-color-5: #f0f0f0;
+
+            --o-cc1-bg: #3aadaa;
+            --o-cc2-bg: #7c6576;
+            --o-cc3-bg: #dd7777;
+            --o-cc4-bg: #383e45;
+            --o-cc5-bg: #f0f0f0;
+        }
+
+        .o_cc5 {
+            background-color: var(--o-cc5-bg);
+        }
+
+        /* Base shape styles - required for positioning and layout */
+        .o_we_shape {
+            position: absolute !important;
+            top: 0;
+            right: 0;
+            bottom: 0;
+            left: 0;
+            display: block;
+            overflow: hidden;
+            pointer-events: none;
+        }
+
+        /* Connections shapes */
+        .o_we_shape.o_html_builder_Connections_01 {
+            background-image: url("/html_editor/shape/html_builder/Connections/01.svg?c5=%23383e45");
+            background-position: 50% 100%;
+            background-size: 100% auto;
+            background-repeat: no-repeat;
+        }
+
+        .o_we_shape.o_html_builder_Connections_02 {
+            background-image: url("/html_editor/shape/html_builder/Connections/02.svg?c5=%23383e45");
+            background-position: 50% 100%;
+            background-size: 100% auto;
+            background-repeat: no-repeat;
+        }
+
+        /* Non-connection shapes used in tests */
+        .o_we_shape.o_html_builder_Blobs_02 {
+            background-image: url("/html_editor/shape/html_builder/Blobs/02.svg?c1=%23383e45");
+            background-position: 50% 50%;
+            background-size: 100% 100%;
+            background-repeat: no-repeat;
+        }
+
+        .o_we_shape.o_html_builder_Containers_01 {
+            background-image: url("/html_editor/shape/html_builder/Containers/01.svg?c5=%23383e45");
+            background-position: 50% 100%;
+            background-size: 100% auto;
+            background-repeat: no-repeat;
+        }
+
+        .o_we_shape.o_html_builder_Bold_16 {
+            background-image: url("/html_editor/shape/html_builder/Bold/16.svg?c5=%23383e45");
+            background-position: 50% 100%;
+            background-size: 100% auto;
+            background-repeat: no-repeat;
+        }
+    `;
+}
+
+test("Connections shape takes neighbor element's background color", async () => {
+    addBuilderOption(
+        class TestBackgroundOption extends BackgroundOption {
+            static selector = "section";
+            static props = {
+                ...BackgroundOption.props,
+                withColors: { type: Boolean, optional: true },
```

Comment:
You can group all of what you want to executed at the beginning of each test in a `beforeEach`